doc/src/tutorials/arch/multi_mode_ble.rst

```diff
@@ -0,0 +1,303 @@
+
+:numref:`fig_frac_lut_le` shows the physical implemenation of a Fracturable Logic Element (FLE), which conssists of a fracturable 6-input Look-Up Table (LUT), two Flip-flops (FFs) and routing multiplexers to select between combinational and sequential outputs.
+
+.. _fig_frac_lut_le:
+
+.. figure:: frac_lut_le.png
+   :scale: 125%
+   
+   Schematic of a fracturable logic element
+
+The FLE in :numref:`fig_frac_lut_le` can operate in two different modes: (a) dual 5-input LUT mode (see :numref:`fig_frac_lut_le_dual_lut5_mode`); and (b) single 6-input LUT mode (see :numref:`fig_frac_lut_le_lut6_mode`). Note that each operating mode does not change the physical implementation of FLE but uses part of the programmable resources.
+
+.. _fig_frac_lut_le_dual_lut5_mode:
+
+.. figure:: frac_lut_le_dual_lut5_mode.png
+   :scale: 125%
+   
+   Simplified organization when the FLE in :numref:`fig_frac_lut_le` operates in dual 5-input LUT mode
+
+.. _fig_frac_lut_le_lut6_mode:
+
+.. figure:: frac_lut_le_lut6_mode.png
+   :scale: 125%
+   
+   Simplified organization when the FLE in :numref:`fig_frac_lut_le` operates in 6-input LUT mode
+
+Architecture Description
+------------------------
+
+To accurately model the operating modes of the FLE, we will use the syntax ``<pb_type>`` and ``<mode>`` in architecture description language.
+
+.. code-block:: xml
+
+   <!-- Multi-mode Fracturable Eogic Element definition begin -->
+   <pb_type name="fle" num_pb="10">
+     <input name="in" num_pins="6"/>
+     <output name="out" num_pins="2"/>
+     <clock name="clk" num_pins="1"/>
+
+     <!-- Dual 5-input LUT mode definition begin -->
+     <mode name="n2_lut5">
+       <!-- Detailed definition of the dual 5-input LUT mode -->
+     </mode>     
+     <!-- Dual 5-input LUT mode definition end -->
+
+     <!-- 6-input LUT mode definition begin -->
+     <mode name="n1_lut6">
+       <!-- Detailed definition of the 6-input LUT mode -->
+     </mode>     
+     <!-- 6-input LUT mode definition end -->
+   </pb_type>
+
+In the above XML codes, we define a ``<pb_type>`` for the FLE by following the port organization in :numref:`fig_frac_lut_le`.
+Under the ``<pb_type>``, we create two modes, ``n2_lut5`` and ``n1_lut6``, corresponding to the two operating modes as shown in :numref:`fig_frac_lut_le_dual_lut5_mode` and :numref:`fig_frac_lut_le_lut6_mode`.
+Note that we focus on operating modes here, which are sufficient to perform architecture evaluation.
+
+Under the dual 5-input LUT mode, we can define ``<pb_type>`` and ``<interconnect>`` to model the schematic in :numref:`fig_frac_lut_le_dual_lut5_mode`.
+
+.. code-block:: xml
+
+   <!-- Dual 5-input LUT mode definition begin -->
+   <mode name="n2_lut5">
+     <pb_type name="lut5inter" num_pb="1">
+       <input name="in" num_pins="5"/>
+       <output name="out" num_pins="2"/>
+       <clock name="clk" num_pins="1"/>
+       <pb_type name="ble5" num_pb="2">
```

Comment:
Should probably have a comment here, calling out `num_pb=2` as why this models two 5-LUT + FF pairs